client-lib/src/detector/tlsh_detector.rs

```diff
@@ -77,43 +91,112 @@ impl DetectorProvider for SimpleTLSHDetectorProvider {
         configuration: &HashMap<String, Box<dyn Any>>,
         system_database: Arc<Mutex<SystemDatabase>>,
     ) -> Box<dyn Detector> {
-        let threshold = if let Some(threshold) = configuration.get("threshold") {
-            let Some(threshold) = threshold.downcast_ref::<i64>() else {
-                panic!("invalid threshold config")
+        let mut system_database = system_database.lock().unwrap();
+        let comparator = if let Some(object) =
+            system_database.get_object::<ColoredTLSHWithDistanceObject>(0x0003)
+        {
+            let database = DistancedTLSHDatabase::new(object);
+            CompareAgainstAllDetector::new(
+                Box::new(database),
+                Box::new(move |hash, stored_hash| {
+                    let diff = stored_hash.diff(hash);
+                    if diff < stored_hash.detection_distance() as i32 {
+                        debug!(
+                            "TLSH below threshold: {} < {}",
+                            diff,
+                            stored_hash.detection_distance()
+                        );
+                        return true;
+                    }
+                    false
+                }),
+            )
+        } else if let Some(legacy_object) = system_database.get_object::<ColoredTLSHObject>(0x0002)
+        {
+            warn!("using legacy database format, please update the database");
+            let database = LegacyTLSHDatabase::new(legacy_object);
+            let threshold = if let Some(threshold) = configuration.get("threshold") {
+                let Some(threshold) = threshold.downcast_ref::<i64>() else {
+                        panic!("invalid threshold config")
+                    };
+                *threshold as i32
+            } else {
+                40
             };
-            *threshold as i32
+            CompareAgainstAllDetector::new(
+                Box::new(database),
+                Box::new(move |hash, stored_hash| {
+                    let diff = stored_hash.diff(hash);
+                    if diff < threshold {
+                        debug!("TLSH below threshold: {} < {threshold}", diff);
+                        return true;
+                    }
+                    false
+                }),
+            )
         } else {
-            40
+            panic!("no usable object found in database. Please update to a later version")
```

Comment:
I think this is ambiguous and we should specify here that the database should be updated, not `simbiota`.